Lib/http/server.py

```diff
@@ -491,6 +494,9 @@ def send_error(self, code, message=None, explain=None):
             body = content.encode('UTF-8', 'replace')
             self.send_header("Content-Type", self.error_content_type)
             self.send_header('Content-Length', str(len(body)))
+        if extra_headers is not None:
+            for (keyword, value) in extra_headers:
+                self.send_header(keyword, value)
```

Comment:
```suggestion
        if extra_headers is not None:
            for name, value in extra_headers:
                self.send_header(name, value)
```

Do we need to document that this would overwrite `Content-Type` and `Content-Length` headers or not? or should we skip them?

Reply:
I think the "extra" in the parameter name "extra_headers" already indicates that the incoming headers should not duplicate the obviously existing headers - including `Content-Type` and `Content-Length`.

The caller also has no reason to specify `Content-Length` as the body is generated and returned within `send_error` method. No caller knows its length.

so IMO additional documentation or code assertions may seem somewhat redundant here.

Reply:
> obviously existing 

One issue I see is the definition of "obviously existing" headers. But I'm fine with not specifying them since sending a custom `Content-Type` for a custom `explain` message is borderline API misuse.

Reply:
Considering that neither `explain` nor `extra_headers` are commonly used fields, such misuse should not be common. Maybe I'll opt to keep it the way it is now unless someone opens a new issue to complain about this.